app/controllers/api/v1/resources_controller.rb

```diff
@@ -0,0 +1,60 @@
+# FIXME extend ActionController::Metal instead as some mixins are not needed
+class Api::V1::ResourcesController < ApplicationController
+  include Pagination
+
+  before_filter :authenticate_api_key!
+  before_filter :require_allowed_model
+
+  # FIXME find a way to document api calls
+
+  def index
+    filter_params = params[model_name.singularize].presence
+
+    resources = filter_params ? model_klass.filter(filter_params) : model_klass.all
```

Comment:
I think there are API specs which would fail for these classes but I am adding a spec to make that requirement explicit.